bopis-mixed-cart-pos/extensions/pickup-and-delivery-options/src/Pickup.jsx

```diff
@@ -0,0 +1,599 @@
+                  setDialogVisible(true);
+                  console.log("Ship option selected and saved.");
+                }}
+              />
+            )}
+            {selectedOption === "Store Pickup Together" && (
+              <Button
+                title="Proceed"
+                type="primary"
+                onPress={() => {
+                  api.navigation.navigate("StorePickup");
+                }}
+              />
+            )}
+            {selectedOption === "Store Pickup Separate" && (
+              <Button
+                title="Proceed"
+                type="primary"
+                onPress={() => {
+                  api.navigation.navigate("StorePickupSeparate");
+                }}
+              />
+            )}
+            <ScrollView>
+              <RadioButtonList
+                items={[
+                  "Ship",
+                  "Store Pickup Together",
+                  "Store Pickup Separate"
+                ]}
+                onItemSelected={(item) => setSelectedOption(item)}
+                initialSelectedItem={selectedOption}
+              />
+            </ScrollView>
+          </Stack>
+        )}
+        {dialogVisible &&
+        <Dialog
+          isVisible={dialogVisible}
+          title="Success"
+          content={dialogContent}
+          actionText="OK"
+          onAction={() => {
+            setDialogVisible(false);
+            api.navigation.dismiss();
+          }}
+          showSecondaryAction={false}
+          type="alert"
+        />
+        }
+      </Screen>
+
+      <Screen
+        name="StorePickup"
+        title="Store Pickup"
+        onNavigateBack={() => {
+          api.navigation.navigate("DeliveryOptions");
+        }}
+        onNavigate={() => {
+          setZipCode("");
+          setAvailableStores([]);
+          setZipCodeError("");
+          setSelectedStore(null);
+          setShowNoStoresMessage(false);
+          setDialogContent("");
+          setDialogVisible(false);
+        }}>
+        <TextField
+          label="Search Available Pickup Stores"
+          value={zipCode}
+          onChange={(value) => {
+            if (zipCodeError && value.length > 0) {
+              setZipCodeError("");
+            }
+            setZipCode(value);
+          }}
+          error={zipCodeError}
+          placeholder="Enter Zip Code"
+        />
+        <Button
+          title="Find Stores"
+          type="basic"
+          onPress={fetchStores}
+          isLoading={loading}
+        />
+        <ScrollView>
+          {availableStores.length > 0 ? (
+            <RadioButtonList
+              items={availableStores.map((store) => `${store.label} (${store.distance})`)}
+              onItemSelected={(item) => {
+                const [label, distance] = item.split(" (");
+                const store = availableStores.find((store) => store.label === label);
```

Comment:
We should not add much logic in the template and should define a function to handle this.